stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -622,6 +638,7 @@ impl NakamotoBlockHeader {
             parent_block_id,
             tx_merkle_root: Sha512Trunc256Sum([0u8; 32]),
             state_index_root: TrieHash([0u8; 32]),
+            timestamp: get_epoch_time_secs(),
```

Comment:
Can you add to the docstring that this is setting the `timestamp` to roughly `Instant::now()`?

I also think that this probably isn't what we want to do-- this will set the timestamp to the start time of mining the block: shouldn't this be set at roughly the end of mining the block?

Reply:
Right, this sets it to the start time. I was thinking that at least the start time is more consistent than the end time, because we can't set it to the time just before broadcasting, since the signers will have already signed, so setting it at the start is better than setting it in the middle (after block built, before signers signed).

But maybe setting it as late as possible makes more sense. My opinion on that is weakly held.

Reply:
Yeah, I don't have a strong view on this -- this seems fine to me.